pandas/io/parsers/base_parser.py

```diff
@@ -125,12 +121,19 @@ def __init__(self, kwds) -> None:
         self.index_names: Sequence[Hashable] | None = None
         self.col_names: Sequence[Hashable] | None = None
 
-        self.parse_dates = _validate_parse_dates_arg(kwds.pop("parse_dates", False))
-        self._parse_date_cols: Iterable = []
+        parse_dates = kwds.pop("parse_dates", False)
+        if parse_dates is None or lib.is_bool(parse_dates):
+            parse_dates = bool(parse_dates)
+        elif not isinstance(parse_dates, list):
+            raise TypeError(
+                "Only booleans and lists are accepted "
+                "for the 'parse_dates' parameter"
+            )
+        self.parse_dates: bool | list = parse_dates
+        self._parse_date_cols: set = set()
```

Comment:
Not a blocker but I find this strange that mypy doesn't infer the type of the variable is a set from the rhs of the assignment

Reply:
Yeah without this annotation I get `error: Need type annotation for "_parse_date_cols" (hint: "_parse_date_cols: Set[<type>] = ...")`